oak_attestation_verification/src/amd.rs

```diff
@@ -0,0 +1,199 @@
+    );
+
+    let verifying_key = {
+        let pubkey_info = signer
+            .tbs_certificate
+            .subject_public_key_info
+            .owned_to_ref();
+        let pubkey = RsaPublicKey::try_from(pubkey_info)
+            .map_err(|_err| anyhow::anyhow!("could not parse RSA public key"))?;
+        rsa::pss::VerifyingKey::<Sha384>::new(pubkey)
+    };
+
+    let message = signee
+        .tbs_certificate
+        .to_der()
+        .map_err(|_err| anyhow::anyhow!("could not extract message to verify RSA signature"))?;
+    let signature = Signature::try_from(signee.signature.raw_bytes())
+        .map_err(|_err| anyhow::anyhow!("could not extract RSA signature"))?;
+
+    Ok(verifying_key
+        .verify(&message, &signature)
+        .map_err(|_err| anyhow::anyhow!("signature verification failed"))?)
+}
+
+// Currently unused, use `pub` only to disable the warning.
+pub fn product_name(cert: &Certificate) -> anyhow::Result<String> {
+    let exts = cert
+        .tbs_certificate
+        .extensions
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("could not get extensions from cert"))?;
+    let pn_ext = exts
+        .iter()
+        .find(|&ext| ext.extn_id == PRODUCT_NAME_OID)
+        .ok_or_else(|| anyhow::anyhow!("no product name found in cert"))?;
+    String::from_utf8(pn_ext.extn_value.as_bytes().to_vec())
+        .map_err(|_utf8_err| anyhow::anyhow!("failed to read product name"))
+}
+
+fn chip_id(cert: &Certificate) -> anyhow::Result<[u8; 64]> {
+    let exts = cert
+        .tbs_certificate
+        .extensions
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("could not get extensions from cert"))?;
+    let chip_id_ext = exts
+        .iter()
+        .find(|&ext| ext.extn_id == CHIP_ID_OID)
+        .ok_or_else(|| anyhow::anyhow!("no chip ID found in cert"))?;
+    let chip_id = chip_id_ext.extn_value.as_bytes().to_vec();
+    anyhow::ensure!(chip_id.len() == 64, "length of chip ID value is not 64");
+
+    // Copy into array of fixed length.
+    let mut result = [0; 64];
+    result.copy_from_slice(&chip_id[0..64]);
+    Ok(result)
+}
+
+fn tcb_version(vcek: &Certificate) -> anyhow::Result<TcbVersion> {
+    let mut tcb = TcbVersion::new_zeroed();
+    for ext in vcek
+        .tbs_certificate
+        .extensions
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("could not get extensions from VCEK cert"))?
+    {
+        // What's the appropriate way of extracting u8 from OctetString?
+        let arr = ext.extn_value.as_bytes();
+        let last = arr.len() - 1;
+        if ext.extn_id == BL_OID {
+            tcb.boot_loader = arr[last];
+        } else if ext.extn_id == TEE_OID {
+            tcb.tee = arr[last];
+        } else if ext.extn_id == SNP_OID {
+            tcb.snp = arr[last];
+        } else if ext.extn_id == UCODE_OID {
+            tcb.microcode = arr[last];
+        }
+    }
+    Ok(tcb)
+}
+
+pub fn verify_attestation_report_signature(
```

Comment:
Should this return the additional data perhaps? (after validating it is correctly signed)
Or do we expect callers to separately extract that for their own use cases?

Reply:
This is coming up, not clear to me what exactly they would need. We don't **really** look inside the attestation report so far (on top of comparing the debug flag against reference values), apart from verifying that it is appropriately signed.